### ISSUE TITLE
reportMissingImports

### ISSUE THREAD

**Issue by omidshojaee** [mozilla/django-csp#240]
Here's the output of ```pip freeze``` command:

```
(venv) PS D:\Projects\Django\artanetwork.com> pip freeze
asgiref==3.8.1
Django==5.1
django-filebrowser-no-grappelli==4.0.2
django-jalali==7.0.0
django-tinymce==4.1.0
django_csp==3.8
jalali_core==1.0.0
jdatetime==5.0.0
pillow==10.4.0
psycopg==3.2.1
psycopg-binary==3.2.1
six==1.16.0
sqlparse==0.5.1
typing_extensions==4.12.2
tzdata==2024.1
```

The ```settings.py``` file:

```
# Application definition

INSTALLED_APPS = [
    'filebrowser',
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'csp',
    'django_jalali',
    'tinymce',
    'accounts.apps.AccountsConfig',
    'homepage.apps.HomepageConfig',
]

MIDDLEWARE = [
    'django.middleware.security.SecurityMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
    'csp.middleware.CSPMiddleware',
]
```

Still, I can't import ```csp.constants``` and Pylance is giving me this error: ```Import "csp.constants" could not be resolved Pylance(reportMissingImports)```

And I can confirm that my configuration is not loaded because the browser tells me ```default-src self``` is in effect while I've set it to NONE.

```
from csp.constants import NONE, SELF

CONTENT_SECURITY_POLICY = {
    "DIRECTIVES": {
        "default-src": [NONE],
    },
}
```

**Comment by janwinsemius** [comment]
I had same issue. Seems there is something wrong woth the build. Some of the files are missing, amongst which constants.py 
For now you could do:
pip install git+https://github.com/mozilla/django-csp.git
and works fine

**Comment by stevejalim** [comment]
Thanks for the reports @omidshojaee and @janwinsemius 

The constants are available in 4.0, which is a pre-release version, but not in 3.8, which is the default current release.

This is why installing from `main` on the git repo works, but installing the default latest release (3.8) from pypi does not.

The docs for 3.8 are here, which don't mention the new `constants` https://django-csp.readthedocs.io/en/3.8/configuration.html

The default 'latest' docs do mention it, which I agree is a little confusing and is something we can try to tidy up.

That said, 4.0 shouldn't be far away from release now anyway. 

In the meantime, If you want to install the very latest version (4.0 beta) that is at https://pypi.org/project/django-csp/4.0b1/ which you can install with `pip install django-csp==4.0b1`

Does that solve your problems? I hope so

**Comment by omidshojaee** [comment]
Thanks for clarification. As I mentioned, even without constants, the browser reports ```default src self``` is in effect while I've set it to ```none```.

Now I understand that this is because I use the new ```CONTENT_SECURITY_POLICY``` format. I changed it to the old ```CSP_DEFAULT_SRC``` format and it works because my installed version is 3.8 but the ```latest``` documentation is about 4.0.

Thanks again.
